source/containers.tex

```diff
@@ -6395,18 +6395,17 @@
 
 \indextext{\idxcode{array}!zero sized}%
 \pnum
-\tcode{array} shall provide support for the special case \tcode{N == 0}.
+In the case that \tcode{N} equals \tcode{0},
+\tcode{begin()} equals \tcode{end()} and
+the return value of \tcode{data()} is unspecified.
```

Comment:
This is a normative change, not editorial, as it gives up the requirement that different empty array objects return iterators with a different value.  That point may be deemed moot as it is observed only by comparing pointers from different ranges.

Reply:
You could observe a difference if the iterator was of pointer type and `std::less` compared iterators from two empty arrays. The intent of the current wording seems to be that the iterators from two separate arrays cannot be equivalent according to the total ordering of pointers.

Reply:
> The intent of the current wording seems to be that the iterators from two separate arrays cannot be equivalent according to the total ordering of pointers.

My reading is that this paragraph intended to guarantee that
- an iterator obtained from `array<T, 0>` object and a value-initialized iterator never compare equal; and
- two iterators obtained from two different `array<T, 0>` objects never compare equal.

And without such guarantees the comparison has (or can have, as decided by the library implementation) undefined behavior ([[iterator.concept.forward]/2](https://eel.is/c++draft/iterator.concept.forward#2), [[forward.iterators]/2](https://eel.is/c++draft/forward.iterators#2)).

However, it is not clear whether the requirement on unique value overrides the specification for forward iterators, and implementations now consistently return value-intialized iterators (libc++ was an exception and conformed to such guarantees, but is not now). I think we should open an LWG issue to drop "unique value".